### ISSUE TITLE
False negatives in multiple rules when linting selectors that only contain a prefix

### ISSUE THREAD

**Issue by undsoft** [angular-eslint/angular-eslint#2768]
The following tests will not report an error, as they should:

**pipe-prefix**
```ts
convertAnnotatedSourceToFailureCase({
  description: 'should fail when Pipe has no selector after the prefix',
  annotatedSource: `
      @Pipe({
        name: 'ng'
              ~~~~
      })
      class Test {}
    `,
  messageId,
  options: [{ prefixes: ['ng'] }],
  data: { prefixes: '"ng"' },
}),
```

**component-selector** and **directive-selector**
```ts
convertAnnotatedSourceToFailureCase({
  description: `should fail if a selector has nothing, but the prefix in camelCase`,
  annotatedSource: `
    @Component({
      selector: 'app'
                ~~~~~
    })
    class Test {}
    `,
  messageId: messageIdStyleFailure,
  options: [{ type: 'element', prefix: 'app', style: 'camelCase' }],
  data: { style: 'camelCase' },
}),
```
